src/hotspot/os/linux/os_linux.cpp

```diff
@@ -927,6 +927,12 @@ bool os::create_thread(Thread* thread, ThreadType thr_type,
   }
   assert(is_aligned(stack_size, os::vm_page_size()), "stack_size not aligned");
 
+  // Add an additional page to the stack size to reduce its chances of getting large page aligned
+  // so that the stack does not get backed by a transparent huge page.
+  if (stack_size >= os::Linux::default_large_page_size() && is_aligned(stack_size, os::Linux::default_large_page_size())) {
+    stack_size += os::vm_page_size();
+  }
```

Comment:
Do we need to watch for a zero default size here so that we don't expand all stacks by a page?

Reply:
Added a check for zero default page size. Please review.